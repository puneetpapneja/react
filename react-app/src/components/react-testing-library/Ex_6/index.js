import React from 'react';

export default class Ex_2 extends React.Component
{

    render(){
        return (
            <div>
                <h1>Example 6</h1>
                <h2>Snapshot testing</h2>
                <b>Need to install react-test-renderer package</b>
            </div>
        )
    }

}