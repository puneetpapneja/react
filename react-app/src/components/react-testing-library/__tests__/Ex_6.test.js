import Ex_6 from '../Ex_6'
import renderer from 'react-test-renderer';

describe('Ex_6 component testing',()=>{
    test('render Example 6',()=>{
        //arrange
        const tree = renderer.create(<Ex_6/>).toJSON();
    
        //act 
        //nothing
    
        //assert        
        expect(tree).toMatchSnapshot();    
    })
})
