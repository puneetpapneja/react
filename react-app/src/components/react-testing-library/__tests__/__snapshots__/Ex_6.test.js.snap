// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ex_6 component testing render Example 6 1`] = `
<div>
  <h1>
    Example 6
  </h1>
  <h2>
    Snapshot testing
  </h2>
  <b>
    Need to install react-test-renderer package
  </b>
</div>
`;
